src/libraries/SwapMath.sol

```diff
@@ -74,13 +74,13 @@ library SwapMath {
             }
 
             // cap the output amount to not exceed the remaining output amount
-            if (!exactIn && amountOut > uint256(-amountRemaining)) {
-                amountOut = uint256(-amountRemaining);
+            if (!exactIn && amountOut > uint256(amountRemaining)) {
+                amountOut = uint256(amountRemaining);
             }
 
             if (exactIn && sqrtRatioNextX96 != sqrtRatioTargetX96) {
                 // we didn't reach the target, so take the remainder of the maximum input as fee
-                feeAmount = uint256(amountRemaining) - amountIn;
+                feeAmount = uint256(-amountRemaining) - amountIn;
```

Comment:
I think I'm not 100% understanding this line. If we have't reached the target the feeAmount is the amount left to swap minus the amount already swapped?

Also why does this case only happen in the exactIn case?

Reply:
> If we have't reached the target the feeAmount is the amount left to swap minus the amount already swapped?

If we havent reached the target (which is the essentially maximum slippage where the trade has to stop), that means that all of the `amountRemaining` has successfully be traded into output tokens. The case this happens, the function follows the following flow...
- calculate `amountRemainingLessFee` which is just the amount to be traded, minus the LP fee
- calculates the `sqrtRatioNextX96` that does get reached by this amount of input tokens
- calculate the exact `amountIn` needed to trade to that next sqrtRatio using `getAmount0Delta`. Due to rounding of calculations this could be slightly different to the `amountRemainingLessFee` that it was going to trade before
- now the true LP fee amount needs to be calculated, which is the difference between the total amount to be traded `amountRemaining` (which is now negative so needs to be flipped), and `amountIn` which is the amount thats actually being swapped. This means that the fee is rounded in the LP's favour.

> Also why does this case only happen in the exactIn case?

LP fees are taken on input tokens. For exactIn, the calculation makes sure that the final amountIn including fees is amountRemaining. For non-exactIn the fee amount can just be calculated as `feePips`% of the amountIn, again rounded in the LP's favour. Theres not a need to make sure that the total `feeAmount + amountIn == amountRemaining` because its not exactIn so the "exact" part stops mattering